### ISSUE TITLE
Environment variable being ignored when using Docker Compose

### ISSUE THREAD

**Issue by x80486** [apimastery/APISimulator#1]
I'm using API Simulator `1.7`, Docker `19.03.13-ce, build 4484c46d9d`, and Docker Compose `1.27.4`.

Im trying to setup the logging using `APISIMULATOR_LOG_FILE`, but it looks like it doesn't have any impact when I start the container.

This is my `docker-compose.yml` file:

```yaml
version: "3.8"

services:
  simulator:
    command: ["apisimulator", "start", "/apisims/stubs"]
    container_name: api-simulator
    environment:
      APISIMULATOR_LOG_FILE: config/log4j2.xml
    image: docker.io/apimastery/apisimulator:1.7     # https://registry.hub.docker.com/r/apimastery/apisimulator/
    networks:
      - global-network
    ports:
      - "6090:6090"
    restart: on-failure
    user: "${USER}:${GROUP}"
    volumes:
      - "${PWD}/stubs:/apisims/stubs"

networks:
  global-network:
    driver: bridge
    name: api-simulator-network
```

...and this is my `log4j2.xml` configuration file:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Configuration monitorInterval="30">
  <Properties>
    <Property name="log-pattern">%d{MM-dd-yyyy HH:mm:ss.SSS} |- %highlight{%5p}{trace=blue, debug=green, info=green, warn=yellow, error=red, fatal=red} in %style{%C{1}:%L}{cyan} [%style{%t}{magenta}] - %m%n</Property>
  </Properties>

  <Appenders>
    <Console name="Console" target="SYSTEM_OUT" follow="true">
      <PatternLayout pattern="${log-pattern}" />
    </Console>
  </Appenders>

  <!-- Logger levels: all, trace, debug, info, warn, error, fatal, off -->
  <Loggers>
    <AsyncLogger name="com.apisimulator" level="debug" additivity="false" includeLocation="true">
      <AppenderRef ref="Console" />
    </AsyncLogger>

    <Root level="warn">
      <AppenderRef ref="Console" />
    </Root>
  </Loggers>
</Configuration>
```

When I start the container, all I can see is this:

```shell
$ docker logs --follow api-simulator 
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/apisimulator/apisimulator-http-1.7/lib/groovy-2.5.10.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

But there is a file `apisimulator.log` that always gets created under my `stubs` directory with a **similar** content all the time:

```shell
2020-10-11T11:13:03.826+0000|INFO|9|main|[+--------------------------+]
2020-10-11T11:13:03.827+0000|INFO|9|main|[| A P I  S i m u l a t o r |]
2020-10-11T11:13:03.827+0000|INFO|9|main|[+--------------------------+]
2020-10-11T11:13:03.828+0000|INFO|9|main|[com.apisimulator.launcher.AbstractLaunchable.init(String[] args): initialization in progress...]
2020-10-11T11:13:03.828+0000|INFO|9|main|[com.apisimulator.launcher.AbstractLaunchable.init(String[] args): {java version:"11.0.8+10", jvm name:"OpenJDK 64-Bit Server VM", jvm version:"11.0.8+10", jvm pid:9}]
2020-10-11T11:13:03.828+0000|INFO|9|main|[com.apisimulator.launcher.AbstractLaunchable.init(String[] args): implementation {title:"HTTP API Simulator",version:"1.7"}]
2020-10-11T11:13:03.828+0000|INFO|9|main|[com.apisimulator.launcher.AbstractLaunchable.init(String[] args): args=[-c, /apisimulator/apisimulator-http-1.7/config, -s, /apisims/stubs/simlets, /apisims/stubs]]
2020-10-11T11:13:03.855+0000|INFO|9|main|[com.apisimulator.config.AppConfigLoader is looking in and under /apisims/stubs/simlets]
2020-10-11T11:13:04.228+0000|INFO|9|main|[com.apisimulator.config.AppConfigLoader loaded 1 file in 373 ms]
2020-10-11T11:13:04.349+0000|INFO|9|main|[com.apisimulator.config.SingleYamlConfigLoader.load(): loaded 2 simlets from '/apisimulator/apisimulator-http-1.7/config/apisim.yaml' in 120 ms]
2020-10-11T11:13:04.495+0000|INFO|9|main|[com.apisimulator.netty.tls.TlsHandlerFactorySupport: ALPN=true; OpenSSL=true]
2020-10-11T11:13:04.749+0000|INFO|9|API-Simulator-Boss-4-1|[com.apisimulator.netty.NettySimulatorServer.start(): listening on 0.0.0.0:6090]
2020-10-11T11:13:04.751+0000|INFO|9|main|[com.apisimulator.launcher.AbstractLaunchable.start(): API Simulator started]
2020-10-11T11:13:04.752+0000|INFO|9|main|[com.apisimulator.launcher.AdminServer: listening on port #6190]
```

**Comment by apisim** [comment]
Hi @x80486 

Are you trying to get the log output to go to the Console? The [logging destination](https://apisimulator.io/docs/1.7/standalone-api-simulator/running-api-simulator.html#logging-destination) can be changed to the Console by passing in the `APISIMULATOR_LOG=stdout` environment variable.

Otherwise, it looks like the location in `APISIMULATOR_LOG_FILE` points to a file outside of the container. Did you try mounting that location as a volume similarly to how `stubs` is mounted?

**Comment by x80486** [comment]
I think I got this one wrong initially; it was the very first day using `API Simulator`. What I was looking for is to get an idea of how close I was when hitting one simulation — if that was being logged anyway.

For instance, `WireMock` and `MockServer` tells you how close you are to match a given mock/stub, stating, for instance: (1) the headers matched, (2) the URL matched, and it looks like you are trying to use `X` mock/stub, but (hypothetically) the query parameters didn't match.

I thought I could see that information by changing the log level, but like I said, I think I got this one wrong.
